.tekton/heavy-test.yaml

```diff
@@ -0,0 +1,197 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  name: heavy-test-client-server
+  annotations:
+    # -----------------------------------------------------------------
+    # TRIGGER LOGIC: Runs only when you comment "/test-heavy" on a PR
+    # -----------------------------------------------------------------
+    pipelinesascode.tekton.dev/on-cel-expression: |
+      event == "issue_comment" &&
+      body.action == "created" &&
+      body.issue.pull_request &&
+      body.comment.body.matches("^/test-heavy")
+      
+    # Load standard tasks from the cluster
+    pipelinesascode.tekton.dev/task: "[git-clone, buildah]"
+    pipelinesascode.tekton.dev/max-keep-runs: "3"
+spec:
+  params:
+    - name: repo_url
+      value: "{{ repo_url }}"
+    - name: revision
+      value: "{{ revision }}"
+    # Create a temporary image tag using the commit hash
+    - name: temp-server-image
+      value: quay.io/ecosystem-appeng/agent-morpheus-rh:pr-test-{{revision}}
+      
+  pipelineSpec:
+    params:
+      - name: repo_url
+      - name: revision
+      - name: temp-server-image
+    workspaces:
+      - name: source
+      - name: basic-auth
+      - name: dockerconfig-ws
+    tasks:
+      # ------------------------------------------------
+      # 1. CLONE THE PR CODE
+      # ------------------------------------------------
+      - name: fetch-repository
+        taskRef:
+          name: git-clone
+        workspaces:
+          - name: output
+            workspace: source
+          - name: basic-auth
+            workspace: basic-auth
+        params:
+          - name: URL
+            value: $(params.repo_url)
+          - name: REVISION
+            value: $(params.revision)
+
+      # ------------------------------------------------
+      # 2. BUILD THE SERVER (From the PR code)
+      # ------------------------------------------------
+      - name: build-server-image
+        runAfter:
+          - fetch-repository
+        taskRef:
+          name: buildah
+        workspaces:
+          - name: source
+            workspace: source
+          - name: dockerconfig
+            workspace: dockerconfig-ws
+        params:
+          - name: IMAGE
+            value: $(params.temp-server-image)
+          - name: DOCKERFILE
+            value: ./Dockerfile
+          - name: CONTEXT
+            value: .
+          - name: TLSVERIFY
+            value: "false" 
+
```

Comment:
@RedTanny Why don't we just use the existing built image tag from fron the on pull request pipelinerun `on-pr-commit-sha-digest` ?